src/status_im/contexts/wallet/wallet_connect/events.cljs

```diff
@@ -141,8 +141,11 @@
 (rf/reg-event-fx
  :wallet-connect/approve-session
  (fn [{:keys [db]}]
-   (let [web3-wallet          (get db :wallet-connect/web3-wallet)
+   (let [theme                (:theme db)
+         web3-wallet          (get db :wallet-connect/web3-wallet)
```

Comment:
Just sharing an idea which I find cheap to do during development: we can extract behavior out of events into private functions right alongside the event when the code is too west-oriented (too many let bindings). Which I find to be the case of this event. This alleviates the mental overhead of understanding the code and promotes REPL usage because inspecting and playing with event handlers is harder, e.g. need to dispatch, need to inject a dummy db, need to see other unrelated re-frame effects in the resulting map, and so on.

Just food for thought :)

Reply:
Yes this would be good to do, but a bit hesitant to refactor things now. What do you say we have a look at the wallet-connect ns after the release candidate and identify potential improvements?

Reply:
Absolutely @clauxx. This is nothing more than a mindset, a place where a Clojure developer might want to be to promote and facilitate REPL usage.